CMakeLists.txt

```diff
@@ -20,6 +20,9 @@ cmake_minimum_required(VERSION 3.16)
 find_package(iceoryx_hoofs QUIET)
 find_package(iceoryx_posh QUIET)
 
+# Link iceoryx_hoofs and iceoryx_posh libraries statically
+set(BUILD_SHARED_LIBS OFF)
+
```

Comment:
I would remove this to give the user a flexible choice if one wants to link static/dynamic.
Setting this to off will always force to link static even one set `-DBUILD_SHARED_LIBS=ON` as CMake parameter from the outside.

```suggestion
```

Reply:
I can see your point. However, I suppose most bindings or examples in external repo like `iceoryx-dds-gateway` or `iceoryx-automotive-soa` will be better off with static linkage. IMHO that should be the default way. Of course, users are free to change it, but most users might not be aware that this option exists and will forget to set it from the outside like I did last week.

Reply:
Doesn't it work with something like `option(BUILD_SHARED_LIBS "Help text" OFF)`?

Reply:
@elBoberido That works, too, of course. But as this is an `option` offered by both `iceoryx_hoofs` and `iceoryx_posh` I thought setting the option here would make more sense than shadowing it.

Reply:
@mossmaurice yes, but then it can be set at the command line like with iceoryx_meta and it's easy to switch between static and shared libs